### PR TITLE
JAMES-2886 extension-jars + jar with dependencies

### PR DESCRIPTION
**Pull request by chibenwa**
This was not clear to me, but things work smoothly with a jar-with-dependencies.

Hence the proposal to solve our kind user issue is to package apache-mailet-crypto in a jar with dependencies.

### CODE REVIEW

**Comment by chibenwa** [comment]
`ObjectStorageBlobsDAOAWSPrefixTest.saveConcurrentlyWithNonPreExistingBucketShouldNotFail » Execution`

test this please

File: server/container/guice/testing/pom.xml [hunk 1]
```diff
@@ -34,5 +34,6 @@
 
     <modules>
         <module>custom-mailets</module>
+        <module>dependency</module>
```
Comment (rouazana) [inline server/container/guice/testing/pom.xml]: do you really have to add it?

Comment (chibenwa) [reply server/container/guice/testing/pom.xml]: To ensure that we don't break compilation randomly?

**Comment by chibenwa** [comment]
Merged

**Pull request closed without merging**